dask/dataframe/io/parquet/arrow.py

```diff
@@ -1123,15 +1123,23 @@ def _collect_dataset_info(
         #          names of partitioned columns.
         #
         partition_obj, partition_names = [], []
-        if (
-            ds.partitioning
-            and ds.partitioning.dictionaries
-            and all(arr is not None for arr in ds.partitioning.dictionaries)
```

Comment:
We actually **do** want to enter this block when one ore more dictionaries are set to `None`, because we now take `None` to mean that we are **not** generating a categorical column. (We previously **required** partitioned columns to be categorical)